src/common/src/util/resource_util.rs

```diff
@@ -170,6 +173,10 @@ pub mod memory {
                 V2_MEMORY_LIMIT_HIERARCHY
             ),
         };
+        let is_max = is_max_in_file_path(&limit_path)?;
+        if is_max {
+            return Ok(get_system_memory());
+        }
         super::util::read_integer_from_file_path(&limit_path)
```

Comment:
That would a double read in common paths. I think we should optimize and reduce it to once.